lib/oli/delivery/sections.ex

```diff
@@ -2666,23 +2669,61 @@ defmodule Oli.Delivery.Sections do
         end)
       end)
 
-    Enum.map(objectives, fn obj ->
-      Map.put(obj, :pages_id, Map.get(objectives_pages_map, obj.objective_resource_id))
-      |> Map.put(
-        :student_proficiency_obj,
-        Map.get(proficiency_per_learning_objective, obj.objective_resource_id, "Not enough data")
-      )
-      |> Map.put(
-        :student_proficiency_subobj,
-        Map.get(
-          proficiency_per_learning_objective,
-          obj.subobjective_resource_id,
-          "Not enough data"
-        )
+    Enum.reduce(objectives, [], fn objective, acc ->
+      # Only consider objectives that belong to a page
+      if Map.get(objectives_pages_map, objective.objective_resource_id) do
+        # Get all the parent objectives of the current objective
+        parent_objectives =
+          Enum.filter(objectives, fn obj ->
+            obj.subobjective_resource_id == objective.objective_resource_id
+          end)
+
+        if length(parent_objectives) == 0 do
```

Comment:
According to Credo:
length(parent_objectives) == 0 is expensive. Prefer Enum.empty?/1 or list == [] (warning:Credo.Check.Warning.ExpensiveEmptyEnumCheck)

Though parent_objectives should never be very long, I would prefer we use `Enum.empty?/1` here instead since it is inside a reduce fn

Reply:
Sure, I'll change it